### PR TITLE
DB-5244: Refactor gatsby-wp templates to use TypeScript

### PR DESCRIPTION
**Pull request by CobyPear**
## What changes were made?
- Convert `gatsby-wp` templates to TypeScript
- Convert handlebars templates in `gatsby-wp` to use tagged template literals
- Moved generator data types to types.ts in order to type the `data` object used for the tagged template literals
- Removed snapshot tests, added a single example snapshot test plus a sample unit test
- Moved all logic from `gatsby-node` to their own files and functions in `lib/createPagesUtils`
- Generated `d.ts` files for gatsby graphql and css module files
- Updated CONTRIBUTING.md to reflect new templates

## Where were the changes made?
`cli`, `gatsby-wp` templates

## How have the changes been tested?
Locally with the watch script.
## Additional information

Don't forget to [add a changeset](https://github.com/pantheon-systems/decoupled-kit-js#generating-a-changeset) if needed!

### CODE REVIEW

**Comment by CobyPear** [comment]
Adding `dompurify` and sanitizing the `dangerouslySetInnerHTML` even though we know it is coming from a trusted source. This should quell the guardrails warnings. This should be added to the next starters too when the time comes.

File: packages/create-pantheon-decoupled-kit/src/templates/gatsby-wp/src/components/paginator.tsx [hunk 1]
```diff
@@ -14,125 +16,119 @@ import * as styles from './paginator.module.css'
...
-		// set the new offset
+		);
+		// If we're on page 1 or greater, set the new offset to
+		// ((currentPageQuery - 1) * itemsPerPage) % totalItems
 		const newOffset =
-			((currentPageQuery > 0 && currentPageQuery - 1) * itemsPerPage) %
-			totalItems
-		setOffset(Number(newOffset))
-	}, [data, offset, itemsPerPage, breakStart, currentPageQuery, totalItems])
+			(Math.max(currentPageQuery - 1, 0) * itemsPerPage) % totalItems;
```
Comment (mitchellmarkoff) [inline packages/create-pantheon-decoupled-kit/src/templates/gatsby-wp/src/components/paginator.tsx:62]: I think something may be off with this new page equation. I am experiencing this regression when trying to access the last page of the pagination example. When first accessing the 10th page, the correct paginated data did show but the URL was still on `/9`. After loading for a bit the URL jumped to `/10` and threw this error. 

https://github.com/pantheon-systems/decoupled-kit-js/assets/39682014/f6371e85-056f-45c3-a332-7bb87058929a

I generated a new `gatsby-wp` starter from canary and was not able to recreate this there.

**Review by mitchellmarkoff — CHANGES_REQUESTED** [review]
This is looking really clean! Not much jumped out to me on this first look over. Aside from the inline comment below, I am noticing a few lint errors on the project instillation that persist when running `pnpm lint:fix` locally. 

Screenshot 2023-06-14 at 3 54 54 PM

**Review by mitchellmarkoff — APPROVED** [review]
Great work on this TS effort! There was a LOT to cover here but I have no more comments after 2 go overs. I'm liking that you broke apart `gatsby-node`, this template is feeling more organized than it's ever been. The starter was generated and working as expected 🕺

**Pull request merged**